inst/qview/qview.R

```diff
@@ -0,0 +1,92 @@
+library(shiny)
+library(tidyverse)
+library(DT)
+
+
+#' queueViewOutput
+#' @export
+queueViewOutput <- function(id) {
+  ns <- NS(id)
+  tableOutput(ns("queue"))
+}
+
+#' #' queueButtonInput
+#' #' @export
+#' queueButtonInput <- function(id, label, job_id) {
+#'   ns <- NS(id)
+#'   actionButton(
+#'     inputId = ns(id),
+#'     label = label,
+#'     icon = icon("stop-circle"),
+#'     job_id = job_id
+#'   )
+#' }
+
+#' queueView
+#' @export
+queueView <- function(input, output, session, queue_obj = rrsq::RSimpleQueue$new()) {
+  output$queue <- renderTable(
+    {
+      invalidateLater(1000, session)
+
+      queueViewFrame <- rrsq::jobs_to_df(queue_obj$get_jobs())
+
+      return(
+        queueViewFrame
+      )
+    }
+  )
+}
+
+
+## Testing ####
+
+action_button_creator <- function(index) {
+  return(
+    as.character(
+      actionButton(
+        as.character(index),
+        glue::glue("index{index}"),
+        onclick = 'Shiny.onInputChange(\"select_button\",  this.id)'
+      )
+    )
+  )
+}
+
+ui <- fluidPage(
+  queueViewOutput("q"),
+  DT::dataTableOutput("test_table"),
+  textOutput("cancel_confirmation")
+)
+
+server <- function(input, output, session) {
+  # whatDoesThisVariableDo <- callModule(queueView, "q", queue_obj = rrsq::RSimpleQueue$new())
```

Comment:
when you've had some more time to play around with this, lets go through this in the next code review, as well as some controls you can enforce in line 67 and 72 around what happens when you have no data and other edge cases.